### PR TITLE
Rewrite of the TimeParser constructor

### PR DESCRIPTION
**Pull request by thiemowmde**
Another attempt to clean up all these different constructor implementations.

This is mostly split from #27.

### CODE REVIEW

File: tests/ValueParsers/TimeParserTest.php [hunk 1]
```diff
@@ -21,7 +22,7 @@ class TimeParserTest extends ValueParserTestBase {
 	 * @deprecated since 0.3, just use getInstance.
 	 */
 	protected function getParserClass() {
-		return 'ValueParsers\TimeParser';
+		throw new \LogicException( 'Should not be called, use getInstance' );
```
Comment (JanZerebecki) [inline tests/ValueParsers/TimeParserTest.php]: If you introduce a backwards compatibility break, then why not remove the function?
This depends on the not yet released 3.0.0 of https://github.com/DataValues/Common but the dependency is not upgraded here.

Comment (thiemowmde) [inline tests/ValueParsers/TimeParserTest.php]: The relevant change was done in #44 and is not part of this PR. When `getInstance` is implemented `getParserClass` is never called.

**Comment by thiemowmde** [comment]
Note that this patch got rebased multiple times. Not much is left here. But I still would love to see these tiny cleanups in master.
